.github/workflows/ci.protocol.yaml

```diff
@@ -0,0 +1,34 @@
+name: Continuous Integration - Protocol
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    paths:
+      - 'protocol/**'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Protocol Tests
+        uses: actions/setup-node@v1
+        with:
+          node-version: '16'
+      - name: Cache Node Modules
+        id: node-modules-cache
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-modules-
```

Comment:
Does this have to match the key above on line 23?

Reply:
No, it's sort of a fallback.

Reply:
Actually might remove that, digging into the docs a bit more.

Reply:
Yes I'll remove it. The documentation is confusing but basically this could restore a cache of node_modules from a different key name, but we really don't want that. We only want to cache a specific version of node_modules based on a key, which is based on a hash of yarn.lock.

Reply:
https://github.com/BeanstalkFarms/Beanstalk/pull/255/commits/b3027157b374bf80e3be5a381e3a5bf5849717c2